docs/reagent-integration.md

```diff
@@ -0,0 +1,112 @@
+# Reagent Integration
+
+Helix can be easily integrated into a Reagent project, which can be useful if you'd like easy access to hooks or less indirection in the creation of your React components.
+
+Calling Helix components from Reagent components requires nothing special; just use the regular Helix component construction form:
+
+``` clojure
+(require '[helix.core :as  h])
+
+(defn reagent-component [a b c]
+  [:div
+   (h/$ helix-component {:foo "bar"})])
+```
+
+This works because Reagent accepts React components in the hiccup returned by components.
+
+Calling Reagent components from Helix components requires the use of `reagent.core/as-element`:
+
+``` clojure
+(require '[reagent.core :as reagent]
+         '[helix.dom :as  h.dom])
+
+(h/defnc helix-component [{:keys [a b c]}]
+  (h.dom/div
+   (reagent/as-element [reagent-component a b c])))
+```
+
+Note that Reagent components can take any number of arguments, whereas Helix components abide by the React rule of components accepting either a single `props` object or a `props` object and a `ref`.
+
+You might want to be able to use Reagent's reactive atoms or Re-frame subscriptions in your Helix components. This is possible with a couple custom hooks:
+
+``` clojure
+(require '["react" :as react]
+         '[re-frame.core :as rf])
+
+(defn use-sub
+  "Subscribe to a re-frame sub.
+  Pass `deps` to make resubscribe when query args change.
+  Borrowed from https://github.com/Lokeh/hooks-demo/blob/master/src/hooks_demo/hooks.cljs"
```

Comment:
Also, this API is not the best. A lot has been learned since that demo was created.

There's no need to pass a `deps` array; rather, you can use the reaction returned by the `query` to determine whether we should resubscribe. This is more complicated to author the hook, but less complicated for consumers.

I think if we're going to include this in the docs, we should have this appear after the `use-deref` below and it should consist of:

```clojure
(defn use-sub
  "Subscribe to a re-frame sub"
  [query]
  (use-deref (rf/subscribe query)))
```